pwa/src/services/filtersToQueryParams.ts

```diff
@@ -28,4 +28,53 @@ export const filtersToQueryParams = (filters: any): string => {
   return params;
 };
 
+export const filtersToUrlQueryParams = (filters: any): string => {
```

Comment:
This function is quite complex, and I'd imagine it will have to stay that way. However, I did try to refactor it to be a bit more concise.

```
export const filtersToUrlQueryParams = (filters: Record<string, any>): string => {
  const cleanedFilters = Object.fromEntries(
    Object.entries(filters).filter(([key]) => !filterKeysToRemove.includes(key)),
  );

  const params = Object.entries(cleanedFilters)
    .map(([key, value]) => {
      if (!value) return null;

      const formattedValue = Array.isArray(value)
        ? value.map((v: string) => v.replace(/\s+/g, "_")).join(`&${key}[]=`)
        : (value as string).replace(/\s+/g, "_");

      return `${Array.isArray(value) ? `${key}[]` : key}=${formattedValue}`;
    })
    .filter(Boolean)
    .join("&");

  return params ? `?${params}` : "";
};
```

I think it does the same, and I've tested it to work. But please do double-check.

Also, we could implement some of the changes in the other function in this file, I think.